### ISSUE TITLE
Gutenboarding: improve domain picker accessibility

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#43556]
> At the Domains step VoiceOver cannot read the “Free” or “Included in plans (price) information. It does say “Recommended”.

![image](https://user-images.githubusercontent.com/87168/85381056-46e56b80-b546-11ea-8cfd-68eda4f69644.png)

**Comment by youbrokesomething** [comment]
I've confirmed that /start/domains will read what domain was searched, any warnings that popped up, and how many domains were found, but a screen reader does not list any of the search results. Tabbing will only find the select buttons and won't read the domain.

**Comment by youbrokesomething** [comment]
This does not happen on /domains/add

**Comment by youbrokesomething** [comment]
Voice Over can read all search result content now!
